api/nodes/serializers.py

```diff
@@ -130,6 +131,128 @@ def update(self, instance, validated_data):
+                                                                                     'user has worked')
+    educational_institutions = ser.ListField(read_only=True, source='schools',
+                                             help_text='An array of dictionaries representing the places the '
+                                                       'user has attended school')
+    social_accounts = ser.DictField(read_only=True, source='social',
+                                    help_text='A dictionary of various social media account dentifiers including '
+                                              'an array of user-defined URLs')
+    bibliographic = ser.BooleanField(help_text='Whether the user will be included in citations for this node or not.  '
+                                               'Required due to issues with field defaulting to false.')
+
+    permission = ser.ChoiceField(choices=['read', 'write', 'admin'], required=False, allow_null=True,
+                                 help_text='Highest permission the user has.  Blank input defaults write permission if '
+                                           'user is being added and to current permission if user is being edited.')
+    links = LinksField({
+        'html': 'absolute_url',
+        'nodes': {
+            'relation': Link('users:user-nodes', kwargs={'user_id': '<pk>'})
+        },
+        'detail': Link('nodes:node-contributor-detail',
+                       kwargs={'node_id': '<node_id>', 'user_id': '<pk>'})
+    })
+
+    class Meta:
+        type_ = 'users'
+
+    def absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        auth = Auth(self.context['request'].user)
+        node = self.context['view'].get_node()
+        user = get_object_or_404(User, validated_data['_id'])
+        # Node object checks for contributor existence but can still change permissions anyway
+        if user in node.contributors:
+            raise exceptions.ValidationError('{} is already a validated contributor'.format(user.username))
+
+        bibliographic = validated_data['bibliographic']
+        permissions = self.get_permissions_list(validated_data['permission']) \
+            if 'permission' in validated_data else ['read', 'write']
+        node.add_contributor(contributor=user, auth=auth, visible=bibliographic, permissions=permissions, save=True)
+        node.reload()
+        user.permission = node.get_permissions(user)[-1]
+        user.bibliographic = node.get_visible(user)
+        user.node_id = node._id
+        return user
+
+    @staticmethod
+    def get_permissions_list(permission):
+        if permission == 'admin':
+            return ['read', 'write', 'admin']
+        elif permission == 'write':
+            return ['read', 'write']
+        elif permission == 'read':
+            return ['read']
+        else:
+            return ['read', 'write']
```

Comment:
Everyone who uses the node object says that. You're not alone. But that's not going to change any time soon.

Also, you can click on a line number, or a range of line numbers, when you're viewing a file and that will add a hashtag to the URL for the file. When following the link, that will highlight those lines. Might as well go ahead and do that here. (I know, it's like a 12 line file, but if you haven't used the feature, it's super-handy)